src/spells.cpp

```diff
@@ -1220,3 +1220,20 @@ bool RuneSpell::executeCastSpell(Creature* creature, const LuaVariant& var, bool
 
 	return scriptInterface->callFunction(3);
 }
+
+bool RuneSpell::canUse(const Player* player) const
+{
+	if (player->hasFlag(PlayerFlag_CannotUseSpells)) {
+		return false;
+	}
+
+	if (player->hasFlag(PlayerFlag_IgnoreSpellCheck)) {
+		return true;
+	}
+
+	if (vocationSpellMap.empty() || hasVocationSpellMap(player->getVocationId())) {
+		return true;
+	}
+
+	return false;
```

Comment:
Let's simplify these statements, also `hasVocationSpellMap` already checks emptiness

```suggestion
	return hasVocationSpellMap(player->getVocationId());
```